assets/app/features/upgrade/controllers/upgrade-openstack-services.controller.js

```diff
@@ -14,82 +14,164 @@
+
         /**
-         *  Validate OpenStackServices required for Cloud 7 Upgrade
+         *  Stop OpenStack services which are non-essential during upgrade
          */
         function stopServices() {
-            vm.openStackServices.running = true;
+            // re-starting after failed backup?
+            if (vm.checks.services.status) {
+                createBackup();
+                return;
+            }
+
+            vm.checks.services.running = true;
 
             upgradeFactory.stopServices()
                 .then(
                     //Success handler. Stop all OpenStackServices successfully:
-                    stopServicesSuccess,
+                    waitForStopServicesToEnd,
                     //Failure handler:
                     stopServicesError
                 );
+        }
 
-            function stopServicesSuccess () {
-                vm.openStackServices.checks.services.status =
-                    vm.openStackServices.valid = true;
-
-                upgradeFactory.createOpenstackBackup()
-                    .then(
-                        //Success handler. Backup OpenStackServices successfully:
-                        createBackupSuccess,
-                        //Failure handler:
-                        createBackupError
-                    ).finally(function() {
-                        vm.openStackServices.completed = true;
-                        // Update openStackServices validity
-                        vm.openStackServices.running = false;
-
-                        vm.openStackServices.valid = vm.openStackServices.checks.backup.status;
-                        // if the backup has finished set the step to complete
-                        if (vm.openStackServices.valid) {
-                            upgradeStepsFactory.setCurrentStepCompleted()
-                        }
-                    });
-            }
+        function stopServicesSuccess(/*response*/) {
+            vm.checks.services.running = false;
+            vm.checks.services.status = true;
+            vm.checks.services.completed = true;
 
-            function stopServicesError (errorOpenStackServicesResponse) {
-                vm.openStackServices.completed = true;
-                vm.openStackServices.running = false;
-                // Expose the error list to openStackServices object
-                vm.openStackServices.errors = errorOpenStackServicesResponse.data.errors;
-            }
+            // trigger backup creation
+            createBackup();
+        }
 
-            function createBackupSuccess (openStackBackupResponse) {
-                vm.openStackServices.checks.backup.status = openStackBackupResponse.data.backup;
-            }
+        function stopServicesError(errorResponse) {
+            vm.checks.services.running = false;
+            vm.checks.services.completed = true
+            // Expose the error list to openStackServices object
+            vm.errors = errorResponse.data;
```

Comment:
Arent we missing here the `UNEXPECTED_ERROR_DATA` stuff that we do in other places?

```
   if (angular.isDefined(errorResponse.data.errors)) {
                            vm.errors = errorResponse.data;
                        } else {
                            vm.errors = UNEXPECTED_ERROR_DATA;
                        }
```